src/components/ModalDialog/ModalDialog.tsx

```diff
@@ -46,22 +47,34 @@ export interface ModalDialogProps extends BaseComponentProps {
   size: ModalDialogSize;
 }
 
+export interface ModalDialogState {
+  isOpen: boolean;
+}
+
 /**
  * A `ModalDialog` is a temporary, modal UI overlay that generally provides contextual app
  * information or requires user confirmation/input.
  */
-export default class ModalDialog extends React.Component<ModalDialogProps> {
+export default class ModalDialog extends React.Component<ModalDialogProps, ModalDialogState> {
   constructor(props: ModalDialogProps) {
     super(props);
   }
 
+  public static getDerivedStateFromProps(props: ModalDialogProps) {
+    return {
+      isOpen: props.isOpen,
+    };
+  }
+
   public render() {
-    const { children, className, isOpen, onDismissed, size } = this.props;
+    const { children, className, onDismissed, size } = this.props;
     const classNames = getClassNames(size);
 
+    this.state.isOpen ? this.startKeyListener() : this.stopKeyListener();
```

Comment:
Don't add listeners as part of render. Perhaps componentDidUpdate.

Reply:
Yup, makes sense.